karamel-core/src/main/java/se/kth/karamel/client/api/KaramelApiImpl.java

```diff
@@ -231,19 +255,194 @@ public SshKeyPair registerSshKeys(String clusterName, SshKeyPair keypair) throws
     return keypair;
   }
 
-
   @Override
   public void registerSudoPassword(String password) {
     ClusterService.getInstance().getCommonContext().setSudoAccountPassword(password);
   }
 
+  @Override
+  public List<OrgItem> listGithubOrganizations() throws KaramelException {
+    return Github.getOrganizations();
+  }
+
+  @Override
+  public List<RepoItem> listGithubRepos(String organization) throws KaramelException {
+    return Github.getRepos(organization);
+  }
+
+  @Override
+  public GithubUser registerGithubAccount(String user, String password) throws KaramelException {
+    return Github.registerCredentials(user, password);
+  }
+
+  @Override
+  public GithubUser loadGithubCredentials() throws KaramelException {
+    return Github.loadGithubCredentials();
+  }
+
+  private void initGithubRepo(String user, String owner, String repo, String description) throws KaramelException {
+    if (owner == null || owner.isEmpty() || owner.compareToIgnoreCase(user) == 0) {
+      Github.createRepoForUser(repo, description);
+    } else {
+      Github.createRepoForOrg(owner, repo, description);
+    }
+  }
+
+  @Override
+  public void commitAndPushExperiment(Experiment experiment) throws KaramelException {
+    String owner = experiment.getGithubOwner();
```

Comment:
Does it make sense to have a ExperimentDesignerService and move the logic of this method and next one and probably some more into that class?